src/libs/antares/study/study.cpp

```diff
@@ -1022,12 +1022,12 @@ bool Study::areaRename(Area* area, AreaName newName)
     // Updating all hydro allocation
     areas.each([&](Data::Area& areait) { areait.hydro.allocation.rename(oldid, newid); });
 
+    ScenarioBuilderUpdater updaterSB(*this);
```

Comment:
The destructor is called later, in particular after `areas.fixOrientationForAllInterconnections`. That's when the sc-builder tmp data is loaded into memory.